addons/web_editor/controllers/main.py

```diff
@@ -270,130 +270,182 @@ def get_image_info(self, image_id=None, xml_id=None):
+
+        # First check the t-call-assets used in the related views
+        url_infos = dict()
+        for v in views:
+            for asset_call_node in etree.fromstring(v["arch"]).xpath("//t[@t-call-assets]"):
+                if asset_call_node.get(resources_type_info['t_call_assets_attribute']) == "false":
+                    continue
+                asset_name = asset_call_node.get("t-call-assets")
+
+                # Loop through bundle files to search for file info
+                _files_data = []
+                for file_info in request.env["ir.qweb"]._get_asset_content(asset_name, {})[0]:
+                    if file_info["atype"] != resources_type_info['mimetype']:
+                        continue
+                    url = file_info["url"]
+
+                    # Exclude library files (see regex above)
+                    if excluded_url_matcher.match(url):
+                        continue
+
+                    # Set files URL filters
+                    custom_files = [
+                        '/website/static/src/scss/user_custom_bootstrap_overridden.scss',
+                        '/website/static/src/scss/user_custom_rules.scss',
+                        '/website/static/src/js/user_custom_javascript.js']
+                    url_from_custom_files = url in custom_files
+
+                    # Check if the file is customized and get bundle/path info
+                    scss_or_js_file_data = self._match_scss_or_js_file_url(url)
+                    if not scss_or_js_file_data:
+                        continue
+                    file_is_customized = scss_or_js_file_data['customized'] is True
+
+                    # Save info according to the filter (arch will be fetched later)
+                    url_infos[url] = scss_or_js_file_data
+                    if resources_type == 'scss':
+                        if only_user_custom_files and len(bundles_restriction) > 0:
+                            if url_from_custom_files or file_is_customized:
+                                _files_data.append(url)
+                        else:
+                            if len(bundles_restriction) == 0:
+                                _files_data.append(url)
+                            else:
+                                if asset_name in bundles_restriction:
+                                    _files_data.append(url)
+                    else:
+                        if url_from_custom_files or file_is_customized:
+                            _files_data.append(url)
+
+                    # File data is returned sorted by bundle, with the bundles names and xmlids
+                    if len(_files_data):
+                        files_data.append([dict(xmlid=asset_name, name=request.env.ref(asset_name).name), _files_data])
+
+        return {'files_data': files_data, 'url_infos': url_infos, 'custom_attachments': custom_attachments}
+
+    def filter_bundles_files_and_fetch(self, data):
```

Comment:
Same goes for this function, is it necessary? Could not this be directly in `load_resources` ?

Reply:
Yes, sure. This is personal, I just find cleaner to have multiple def when functions are very long :smiley:

Reply:
Most of the time yes, but when it is about modifying existing code, it is better to have a minimal diff, so it is easier to review. Also, in my current task I move all of that code in another file and split it in sub-functions... so I will have a hell of a rebase once your task is merged, so it is even more reasons to keep the diff small :)

Reply:
Good to know, I can surely understand :smile: